keanu-project/src/main/java/io/improbable/keanu/vertices/Vertex.java

```diff
@@ -155,40 +146,40 @@ public void setAndCascade(T value) {
      *
      * @param value the value to be observed
      */
+    @Override
     public void observe(T value) {
         this.value = value;
-        this.observed = true;
-    }
-
-    /**
-     * Cause this vertex to observe its own value, for example when generating test data
-     */
-    public void observeOwnValue() {
-        this.observed = true;
+        observation.observe(value);
     }
 
+    @Override
     public void unobserve() {
-        observed = false;
+        observation.unobserve();
     }
 
+    @Override
     public boolean isObserved() {
-        return observed;
+        return observation.isObserved();
+    }
+
+    public boolean matchesObservation() {
```

Comment:
What matches Observation?

Reply:
the name comes from `RejectionSampler.java` (see above) - can you think of a better name for this?